### ISSUE TITLE
liftEither in MutResolver ?

### ISSUE THREAD

**Issue by diskroma** [morpheusgraphql/morpheus-graphql#351]
How can i create a MutResolver with an Either return value ?
Essentially I want a mutation that sends out a subscription notification, if the mutation succeeds.

Something like this : 
```
...
MutResolver
  $ do
    value <- liftEither (resolveJoinLobby' session pipe _id)
    return ([Event [USER] (Content { contentID = 12 })], value)

...

 resolveJoinLobby' :: Maybe User.SessionE
                   -> Pipe
                   -> Text
                   -> IO (Either String (Lobby (IOMutRes USEREVENT)))
``` 

This is similar to one of the examples, but instead of a lift, I would need liftEither, since `    resolveJoinLobby' ` returns a resolver that encapsulates an either type.

It does not compile like that, complaining about some kinds. Here is the error message: 

```
* Couldn't match kind `*'
                 with `morpheus-graphql-0.8.0:Data.Morpheus.Types.Internal.AST.Data.OperationType'
  When matching types
    res0 :: morpheus-graphql-0.8.0:Data.Morpheus.Types.Internal.AST.Data.OperationType
            -> Bool -> (* -> *) -> * -> *
    morpheus-graphql-0.8.0:Data.Morpheus.Types.Internal.Resolving.Core.ResultT
      (Event Channel Content) :: * -> Bool -> (* -> *) -> * -> *
  Expected type: morpheus-graphql-0.8.0:Data.Morpheus.Types.Internal.Resolving.Core.ResultT
                   (Event Channel Content)
                   String
                   'True
                   IO
                   (Lobby (IOMutRes USEREVENT))
    Actual type: res0 o0 'True IO (Lobby (IOMutRes USEREVENT))
* In a stmt of a 'do' block:
    value <- liftEither (resolveJoinLobby' session pipe _id)
  In the second argument of `($)', namely
    `do value <- liftEither (resolveJoinLobby' session pipe _id)
        return ([Event [...] (Content {contentID = 12})], value)'
  In the expression:
    MutResolver
      $ do value <- liftEither (resolveJoinLobby' session pipe _id)
           return ([Event [...] (Content {contentID = 12})], value)ghcmod
```

**Comment by dandoh** [comment]
MutResolver wraps `ResultT event String 'True m ([event], value)` and turns it to `Resolver MUTATION event m value`. 

`LiftEither` is only for `Resolver MUTATION event m` monad, but not `ResultT event String 'True m`, thus the type error.

I think this is the solution:
```haskell
... = do
value <- liftEither (resolveJoinLobby' session pipe _id)
MutResolver $ pure ([Event [USER] (Content { contentID = 12 })], value)

...

 resolveJoinLobby' :: Maybe User.SessionE
                   -> Pipe
                   -> Text
                   -> IO (Either String (Lobby (IOMutRes USEREVENT)))
```

**Comment by nalchevanidze** [comment]
it is bug. `liftEither` should be almost same as `lift` and does not need it's own class. i will fix it

**Comment by nalchevanidze** [comment]
fixed in #356

**Comment by nalchevanidze** [comment]
https://github.com/morpheusgraphql/morpheus-graphql/blob/a2946ddee170264a9d392008c35ea1843e0593a7/examples/src/Server/Sophisticated/API.hs#L146-L149

there is an example
